matsim/src/main/java/org/matsim/core/population/PopulationUtils.java

```diff
@@ -1093,4 +1080,49 @@ public static boolean comparePopulations( Population population1, Population pop
 		boolean result = PopulationUtils.equalPopulation( population1, population2 );
 		return result ;
 	}
+
+	// ---
+
+	public static Object getPersonAttribute( HasPlansAndId person, String key, Population population ) {
+		Object result = null ;
+		if ( person instanceof Attributable ){
+			result = ((Attributable) person).getAttributes().getAttribute( key );
+		}
+		if ( result != null ) {
+			return result ;
+		}
+		if ( population==null ) {
+			log.warn( "population==null (which is ok) but requested attribute not in person.  Probably ok; will return null" ) ;
+			// person.getAttributes().getAttribute( key ) would also return null when attribute is not there; remaining corner case is that it is
+			// actually in population.getPersonAttributes() but that is not available when population is null. kai, may'19
+
+			return null ;
+		}
+		return population.getPersonAttributes().getAttributeDirectly( person.getId().toString(), key ) ;
+	}
+	public static void putPersonAttribute( Person person, String key, Object value ) {
+		person.getAttributes().putAttribute( key, value ) ;
+	}
+	public static Object removePersonAttribute( Person person, String key, Population population ) {
+		Object result1 = person.getAttributes().removeAttribute( key );
+		Object result2 = population.getPersonAttributes().removeAttributeDirectly( person.getId().toString(), key ) ;
+		if( result1==null ) {
+			return result2 ;
+		}
+		if ( result2==null ) {
+			return result1 ;
+		}
+		if ( Objects.equals( result1, result2 ) ) {
+			return result1 ;
+		}
+		throw new RuntimeException("inconsistent object attributes state; don't know what to do.  Try using PopulationUtils" +
+							     ".get/put/...PersonAttribute instead of population.getPersonAttributes... .. kai, " +
+							     "mar'19") ;
+	}
+	@Deprecated  // this command is a bit dangerous, since it might clear someone else's attributes.  Maybe best don't use.  kai, may'19
```

Comment:
What is meant here by "someone else's attributes"? Of another `Person`? Why?

Reply:
Well, if you are done with your personAttributes, and clear them by using this method, then you will (evidently) also clear everybody else's person attributes.  Which means that we may have two codes that work separately, but not when combined.  Thus I think we could decide to not offer such a method.